geth/jail/execcution_policy.go

```diff
@@ -0,0 +1,196 @@
+package jail
+
+import (
+	"context"
+	"encoding/json"
+	"math/big"
+	"time"
+
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/robertkrimen/otto"
+	"github.com/status-im/status-go/geth/common"
+)
+
+// ExecutionPolicy provides a central container for the executions of RPCCall requests for both
+// remote/upstream processing and internal node processing.
+type ExecutionPolicy struct{}
+
+// ExecuteSendTransaction defines a function to execute RPC requests for eth_sendTransaction method only.
+func (ep ExecutionPolicy) ExecuteSendTransaction(manager common.NodeManager, account common.AccountManager, req common.RPCCall, call otto.FunctionCall) (*otto.Object, error) {
+	config, err := manager.NodeConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	if config.UpstreamConfig.Enabled {
+		return ep.ExecuteRemoteSendTransaction(manager, account, req, call)
+	}
+
+	return ep.ExecuteLocalSendTransaction(manager, req, call)
+}
+
+// ExecuteRemoteSendTransaction defines a function to execute RPC method eth_sendTransaction over the upstream server.
+func (ExecutionPolicy) ExecuteRemoteSendTransaction(manager common.NodeManager, account common.AccountManager, req common.RPCCall, call otto.FunctionCall) (*otto.Object, error) {
+	config, err := manager.NodeConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	selectedAcct, err := account.SelectedAccount()
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := manager.RPCClient()
+	if err != nil {
+		return nil, err
+	}
+
+	fromAddr, err := req.ParseFromAddress()
+	if err != nil {
+		return nil, err
+	}
+
+	toAddr, err := req.ParseToAddress()
+	if err != nil {
+		return nil, err
+	}
+
+	// We need to request a new transaction nounce from upstream node.
+	ctx, canceller := context.WithDeadline(context.Background(), time.Now().Add(1*time.Minute))
+	defer canceller()
+
+	var num hexutil.Uint
+	if err := client.CallContext(ctx, &num, "eth_getTransactionCount", fromAddr, "latest"); err != nil {
+		return nil, err
+	}
+
+	nonce := uint64(num)
```

Comment:
There's no way to eliminate this rather than caching nonce on our side. In our case, it means caching and propagating to other client devices.

We may successfully get `nonce` from the upstream and send a transaction and then do it one more time and get a conflict (even in case when requests are sent in sequence) because there's no guarantee that our request will reach the same server next time.

But I've just realized that this could happen even now if the second query reach the same server due to: https://github.com/status-im/status-go/pull/193/files#diff-75b3479b302072c58c0e04b20ad3d633R79
@influx6, we should change "latest" to "pending". Otherwise, a newly sent transaction which is still in a memory pool and not included in block wouldn't count. Please, do.